mirgecom/diffusion.py

```diff
@@ -149,25 +171,26 @@ def __init__(self, value):
         """
         self.value = value
 
-    def get_q_flux(self, discr, alpha, dd, u):  # noqa: D102
+    def get_q_flux(self, discr, var_diff_quad_tag, alpha, dd, u):  # noqa: D102
         u_int = discr.project("vol", dd, u)
         u_tpair = TracePair(dd, interior=u_int, exterior=u_int)
-        return _q_flux(discr, alpha, u_tpair)
+        return _q_flux(discr, var_diff_quad_tag, alpha, u_tpair)
 
-    def get_u_flux(self, discr, alpha, dd, q):  # noqa: D102
-        ones = discr.zeros(q[0].array_context) + 1.
-        bdry_ones = discr.project("vol", dd, ones)
+    def get_u_flux(self, discr, var_diff_quad_tag, alpha, dd, q):  # noqa: D102
+        dd_quad = dd.with_qtag(var_diff_quad_tag)
+        dd_allfaces_quad = dd_quad.with_dtag("all_faces")
+        alpha_int = discr.project("vol", dd, alpha)
         # Compute the flux directly instead of constructing an external q value
         # (and the associated TracePair); this approach is simpler in the
         # spatially-varying alpha case (the other approach would result in a
         # q_tpair that lives in the quadrature discretization, as it involves
         # computing sqrt(alpha); _u_flux would need to be modified to accept such
         # values).
-        flux_weak = alpha*self.value*bdry_ones
-        return discr.project(dd, "all_faces", flux_weak)
+        flux_quad = discr.project(dd, dd_quad, -alpha_int*self.value)
+        return discr.project(dd_quad, dd_allfaces_quad, flux_quad)
 
 
-def diffusion_operator(discr, alpha, boundaries, u):
+def diffusion_operator(discr, alpha, boundaries, u, var_diff_quad_tag=QTAG_NONE):
     r"""
     Compute the diffusion operator.
```

Comment:
Give the operator (including the diffusivity) as TeX.

Reply:
It's there in the lines below that (pydocstyle complains when I make the first sentence of the docstring span more than one line).

Reply:
Ah, missed it!